common/web/gesture-recognizer/src/engine/touchEventEngine.ts

```diff
@@ -63,10 +73,14 @@ export class TouchEventEngine<HoveredItemType, StateToken = any> extends InputEv
     }
   }
 
-  public dropTouchpointWithId(identifier: number) {
-    super.dropTouchpointWithId(identifier);
+  public dropTouchpoint(source: GestureSource<HoveredItemType>) {
+    super.dropTouchpoint(source);
 
-    delete this.safeBoundMaskMap[identifier];
+    for(let key in this.safeBoundMaskMap) {
```

Comment:
Two things:

1. As above, `for..in` is not safe.

```suggestion
    for(const key of Object.keys(this.safeBoundMaskMap)) {
```

2. Particularly when you [delete properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#deleted_added_or_modified_properties) while in the loop.

Finally, have you considered `Map` as it is faster than an Object map apparently.

Reply:
Oh, right, point 2 there.  I'm usually vigilant about catching that; sorry for missing that!